packages/block-library/src/post-content/edit.js

```diff
@@ -45,40 +51,118 @@ function ReadOnlyContent( {
+function EmptyContentPlaceholder( { context, onClose, openInserter } ) {
+	const { postType } = context;
+	const label =
+		'page' === postType
+			? __( 'This page’s content is empty' )
+			: __( 'This post’s content is empty' );
+	return (
+		<Placeholder
+			icon={ icon }
+			label={ label }
+			instructions={ __(
+				'Add your first block or pattern to get started.'
+			) }
+		>
+			<Button variant="primary" onClick={ openInserter }>
+				{ __( 'Choose a pattern' ) }
+			</Button>
 
+			<Button variant="secondary" onClick={ onClose }>
+				{ __( 'Start blank' ) }
+			</Button>
+		</Placeholder>
+	);
+}
+
+function PostContentPlaceholder( { layoutClassNames } ) {
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+	return (
+		<div { ...blockProps }>
+			<p>
+				{ __(
+					'This is the Content block, it will display all the blocks in any single post or page.'
+				) }
+			</p>
+			<p>
+				{ __(
+					'That might be a simple arrangement like consecutive paragraphs in a blog post, or a more elaborate composition that includes image galleries, videos, tables, columns, and any other block types.'
+				) }
+			</p>
+			<p>
+				{ __(
+					'If there are any Custom Post Types registered at your site, the Content block can display the contents of those entries as well.'
+				) }
+			</p>
+		</div>
+	);
+}
+
+function EditableContent( { context = {}, clientId } ) {
+	const { postType, postId } = context;
+	const { selectBlock, insertBlock } = useDispatch( blockEditorStore );
 	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
 		postType,
 		{ id: postId }
 	);
 
-	const entityRecord = useSelect(
+	const { entityRecord, setInserterIsOpened } = useSelect(
 		( select ) => {
-			return select( coreStore ).getEntityRecord(
-				'postType',
-				postType,
-				postId
-			);
+			return {
+				entityRecord: select( coreStore ).getEntityRecord(
+					'postType',
+					postType,
+					postId
+				),
+				setInserterIsOpened:
+					select( blockEditorStore ).getSettings()
+						.__experimentalSetIsInserterOpened,
+			};
 		},
 		[ postType, postId ]
 	);
 
 	const hasInnerBlocks = !! entityRecord?.content?.raw || blocks?.length;
+	const [ hasPlaceholder, setHasPlaceholder ] = useState( ! hasInnerBlocks );
 
-	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
-
-	const props = useInnerBlocksProps(
+	const { children, ...props } = useInnerBlocksProps(
 		useBlockProps( { className: 'entry-content' } ),
 		{
 			value: blocks,
 			onInput,
 			onChange,
-			template: ! hasInnerBlocks ? initialInnerBlocks : undefined,
 		}
 	);
-	return <div { ...props } />;
+	const onClose = () => {
+		setHasPlaceholder( false );
+		const initialBlock = createBlock( 'core/paragraph' );
+		insertBlock( initialBlock, 0, clientId );
+		selectBlock( initialBlock.clientId );
+	};
+
+	const openInserter = () => {
+		setHasPlaceholder( false );
+		setInserterIsOpened( {
+			initialCategory: 'patterns',
```

Comment:
Here I've extended the action value of `setIsInserterOpened`.

It's a core editor store action:

https://github.com/WordPress/gutenberg/blob/d8df29431ae91ac6ece3d9888504e1eae6f0636a/docs/reference-guides/data/data-core-editor.md#setisinserteropened

Reply:
Seems reasonable. We should update the docs for `setIsInserterOpened`.

Reply:
Thanks for confirming 🍺  

I was just about to ping you about this after searching in vain for non-💩 alternatives.